cli/azd/pkg/azdo/project.go

```diff
@@ -0,0 +1,190 @@
+	count := 0
+
+	for !projectCreated {
+		operation, err := operationsClient.GetOperation(ctx, getOperationsArgs)
+		if err != nil {
+			return nil, err
+		}
+
+		if *operation.Status == "succeeded" {
+			projectCreated = true
+		}
+
+		if count >= maxCheck {
+			return nil, fmt.Errorf("error creating azure devops project %s", name)
+		}
+
+		count++
+		time.Sleep(800 * time.Millisecond)
+	}
+
+	project, err := getAzdoProjectByName(ctx, connection, name)
+	if err != nil {
+		return nil, err
+	}
+	return project, nil
+}
+
+// prompts the user for a new AzDo project name and creates the project
+// returns project name, project id, error
+func GetAzdoProjectFromNew(ctx context.Context, repoPath string, connection *azuredevops.Connection, env *environment.Environment, console input.Console) (string, string, error) {
+	var project *core.TeamProjectReference
+	currentFolderName := filepath.Base(repoPath)
+	var projectDescription string = AzDoProjectDescription
+
+	for {
+		name, err := console.Prompt(ctx, input.ConsoleOptions{
+			Message:      "Enter the name for your new Azure DevOps Project OR Hit enter to use this name:",
+			DefaultValue: currentFolderName,
+		})
+		if err != nil {
+			return "", "", fmt.Errorf("asking for new project name: %w", err)
+		}
+		var message string = ""
+		newProject, err := createProject(ctx, connection, name, projectDescription, console)
+		if err != nil {
+			message = err.Error()
+		}
+		if strings.Contains(message, fmt.Sprintf("The following project already exists on the Azure DevOps Server: %s", name)) {
+			console.Message(ctx, fmt.Sprintf("error: the project name '%s' is already in use\n", name))
+			continue // try again
+		} else if strings.Contains(message, "The following name is not valid") {
+			console.Message(ctx, fmt.Sprintf("error: the project name '%s' is not a valid Azure DevOps project Name. See https://aka.ms/azure-dev/azdo-project-naming\n", name))
+			continue // try again
+		} else if err != nil {
+			return "", "", fmt.Errorf("creating project: %w", err)
+		} else {
+			project = newProject
+			break
+		}
+	}
+
+	return *project.Name, project.Id.String(), nil
+}
+
+// return an azdo project by name
+func getAzdoProjectByName(ctx context.Context, connection *azuredevops.Connection, name string) (*core.TeamProjectReference, error) {
+	coreClient, err := core.NewClient(ctx, connection)
+	if err != nil {
+		return nil, err
+	}
+
+	args := core.GetProjectsArgs{}
+	getProjectsResponse, err := coreClient.GetProjects(ctx, args)
+	if err != nil {
+		return nil, err
+	}
+
+	projects := getProjectsResponse.Value
+	for _, project := range projects {
+		if *project.Name == name {
+			return &project, nil
+		}
+	}
+
+	return nil, fmt.Errorf("azure devops project %s not found", name)
+}
+
+// prompt the user to select form a list of existing Azure DevOps projects
+func GetAzdoProjectFromExisting(ctx context.Context, connection *azuredevops.Connection, console input.Console) (string, string, error) {
```

Comment:
`Azdo` is very repetitive in all these function names considering you are already in the `azdo` package.
```suggestion
func GetProjectFromExisting(ctx context.Context, connection *azuredevops.Connection, console input.Console) (string, string, error) {
```

Reply:
That is a good call out thanks. These were originally part of the pipeline package under pkg/commands/pipeline , so they were prefixed with AzDo to differentiate it from the GitHub functions. However, last week I refactored the code to the azdo package, so the extra AzDo is redundant. I'll remove them.

Reply:
done